StreetHawk/Classes/Core/Publish/SHUtils.m

```diff
@@ -869,38 +869,69 @@ + (UIColor *)colorFromHexString:(NSString *)hexString
     {
         return nil;
     }
-    if (hexString.length != 4 //#RGB
-        && hexString.length != 7 //#RRGGBB
-        && hexString.length != 9) //AARRGGBB
-    {
-        return nil;
-    }
-    if (![hexString hasPrefix:@"#"])
-    {
-        return nil;
-    }
-    if (hexString.length == 4)
-    {
-        NSString *red = [hexString substringWithRange:NSMakeRange(1, 1)];
-        NSString *green = [hexString substringWithRange:NSMakeRange(2, 1)];
-        NSString *blue = [hexString substringWithRange:NSMakeRange(3, 1)];
-        hexString = [NSString stringWithFormat:@"#%@%@%@%@%@%@", red, red, green, green, blue, blue];
-    }
-    NSScanner *scanner = [NSScanner scannerWithString:hexString];
-    [scanner setScanLocation:1]; // bypass '#' character
-    unsigned rgbValue = 0;
-    [scanner scanHexInt:&rgbValue];
-    CGFloat red, green, blue, alpha;
-    red = ((rgbValue & 0xFF0000) >> 16)/255.0;
-    green = ((rgbValue & 0xFF00) >> 8)/255.0;
-    blue = (rgbValue & 0xFF)/255.0;
-    if (hexString.length == 7)
-    {
-        alpha = 1.0;
+    CGFloat red = -1;
+    CGFloat green = -1;
+    CGFloat blue = -1;
+    CGFloat alpha = -1;
+    if (hexString.length == 4 //#RGB
```

Comment:
var RGB = 4;
var RRGGBB = 7;
var AARRGGBB = 9;
try no magic numbers

Reply:
Yes. How to get rid of these number in logic?